src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncTargetBlockSelector.scala

```diff
@@ -83,12 +82,22 @@ class FastSyncTargetBlockSelector(
   def tryChooseTargetBlock(receivedHeaders: Map[Peer, BlockHeader]): Unit = {
     log.debug("Trying to choose fast sync target block. Received {} block headers", receivedHeaders.size)
     if (receivedHeaders.size >= minPeersToChooseTargetBlock) {
-      val (mostUpToDatePeer, mostUpToDateBlockHeader) = receivedHeaders.maxBy(_._2.number)
+
+      val (_, mostUpToDateBlockHeader) = receivedHeaders.maxBy(_._2.number)
       val targetBlock = mostUpToDateBlockHeader.number - syncConfig.targetBlockOffset
-      peerEventBus ! Subscribe(MessageClassifier(Set(BlockHeaders.code), PeerSelector.WithId(mostUpToDatePeer.id)))
-      etcPeerManager ! EtcPeerManagerActor.SendMessage(GetBlockHeaders(Left(targetBlock), 1, 0, reverse = false), mostUpToDatePeer.id)
+
+      val peersToAsk = receivedHeaders.toList.sortBy(-_._2.number)
+        .takeWhile { case (_, header) => header.number > targetBlock }
```

Comment:
we should probably make sure that we have enough peers, and if not then reschedule asking for new block